js/pro/cex.js

```diff
@@ -0,0 +1,1216 @@
+        }
+        const messageHash = 'trades';
+        this.trades = stored;
+        client.resolve (this.trades, messageHash);
+    }
+
+    async watchTicker (symbol, params = {}) {
+        /**
+         * @method
+         * @name cex#watchTicker
+         * @see https://cex.io/websocket-api#ticker-subscription
+         * @description watches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+         * @param {string} symbol unified symbol of the market to fetch the ticker for
+         * @param {object} params extra parameters specific to the cex api endpoint
+         * @param {string|undefined} params.method public or private
+         * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        symbol = market['symbol'];
+        const url = this.urls['api']['ws'];
+        const messageHash = 'ticker:' + symbol;
+        const method = this.safeString (params, 'method', 'private'); // default to private because the specified ticker is received quicker
+        let message = {
+            'e': 'subscribe',
+            'rooms': [
+                'tickers',
+            ],
+        };
+        let subscriptionHash = 'tickers';
+        if (method === 'private') {
+            await this.authenticate ();
+            message = {
+                'e': 'ticker',
+                'data': [
+                    market['baseId'], market['quoteId'],
+                ],
+                'oid': this.requestId (),
+            };
+            subscriptionHash = 'ticker:' + symbol;
+        }
+        const request = this.deepExtend (message, params);
+        return await this.watch (url, messageHash, request, subscriptionHash);
+    }
+
+    async watchTickers (symbols = undefined, params = {}) {
+        /**
+         * @method
+         * @name cex#watchTickers
+         * @see https://cex.io/websocket-api#ticker-subscription
+         * @description watches price tickers for multiple markets, statistical calculations with the information calculated over the past 24 hours each market
+         * @param {[string]|undefined} symbols unified symbols of the markets to fetch the ticker for, all market tickers are returned if not assigned
+         * @param {object} params extra parameters specific to the cex api endpoint
+         * @returns {object} an array of [ticker structures]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        await this.loadMarkets ();
+        symbols = this.marketSymbols (symbols);
+        const url = this.urls['api']['ws'];
+        const messageHash = 'tickers';
+        const message = {
+            'e': 'subscribe',
+            'rooms': [
+                'tickers',
+            ],
+        };
+        const request = this.deepExtend (message, params);
```

Comment:
I don't think this implementation works because cex broadcasts one symbol at a time so if it's not the symbol you subscribed to, we return an empty object 🤔

Reply:
this is actually an issue we might have in other exchanges too with `watchTickers`. I proposed here a solution